platform/ios/src/MGLAnnotationView.mm

```diff
@@ -108,6 +117,122 @@ - (void)updateScaleForPitch:(CGFloat)pitch
     }
 }
 
+#pragma mark - Draggable
+
+- (void)setDraggable:(BOOL)draggable
+{
+    [self willChangeValueForKey:@"draggable"];
+    _draggable = draggable;
+    [self didChangeValueForKey:@"draggable"];
+    
+    if (draggable)
+    {
+        [self enableDrag];
+    }
+    else
+    {
+        [self disableDrag];
+    }
+}
+
+- (void)enableDrag
+{
+    if (!_longPressRecognizer)
+    {
+        UILongPressGestureRecognizer *recognizer = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(handleLongPress:)];
+        recognizer.delegate = self;
+        [self addGestureRecognizer:recognizer];
```

Comment:
@1ec5 
Just tested and that doesn't seem to be the case.
- `-enableDrag` creates a gesture recognizer and adds it to the view which holds a strong reference to it. (the property holds a weak ref to the recognizer)
- `-disableDrag` removes the gesture recognizer from the view and since the property only holds a weak ref, it gets nil'd out by `[self removeGestureRecognizer:_longPressRecognizer];`
- Next time `-enableDrag` is called, a new gesture recognizer will be created and added to the view because the property is nil.